### DISCUSSION TITLE
Topic Level Replication

### DISCUSSION THREAD

**Discussion by Apurva007** [apache/pulsar#19682] [Q&A]
Apache Pulsar documentation states that geo-replication can be enabled at namespace or topic level. I am currently testing topic level replication without the whole namespace being replicated.

Setup:
I have 2 test clusters A and B sharing a global ZK. On Cluster A, I created a new namespace "geo" in tenant "public". In the "geo" namespace I created a new persistent partitioned topic "geo-topic". And then I used the following command to setup topic-level replication by running it on cluster A:

bin/pulsar-admin topics set-replication-clusters --clusters A,B public/geo/geo-topic

The command ran successfully, but I don't see any replication occurring. The topic and namespace is also missing on the cluster B. 

Please can someone clarify if I am missing anything or misunderstood the documentation?

**Comment by gauravAshok** [comment]
Please share the pulsar version and the output of `tenants get default` & `clusters list`.

**Reply by Apurva007** [reply]
@gauravAshok  Its a test cluster.
Pulsar version: 2.11.0

`tenants get default` = Tenants does not exist
`clusters list` =
test-A 
test-B

**Reply by Apurva007** [reply]
`tenants get public`=
{
  "adminRoles" : [ ],
  "allowedClusters" : [ "test-A", "test-B" ]
}

**Reply by gauravAshok** [reply]
Then I think the command you meant to run for topic replication is
```shell
bin/pulsar-admin topics set-replication-clusters --clusters test-A,test-B public/geo/geo-topic
```

**Reply by Apurva007** [reply]
Yes. I ran the same command that you have specified. While typing the question I was using example cluster names.

**Reply by gauravAshok** [reply]
ok. :)
Then the only thing I can think of is the pulsar broker configuration

important ones are:
systemTopicEnabled=true
topicLevelPoliciesEnabled=true

Other than the above you should look into the broker logs to get more information. Check the logs when you run "set-replication-clusters" command.

**Reply by Apurva007** [reply]
So in 2.11.0 these configs are enabled by default. So didnt see any issues there. 
Quick question. To set topic level replication, is the parent namespace replication required to be enabled as well?

**Reply by gauravAshok** [reply]
I dont think so. setting this on namespace will probably enable replication for all topics under it.

**Reply by Apurva007** [reply]
Right. I thought the same. But interestingly, the guide in the pulsar docs are not matching the actual results. So was wondering if anyone else had tried this out.
